src/dataDisplay/tableHeaderCell.jsx

```diff
@@ -1,32 +1,360 @@
+        PropTypes.oneOf(columnNumberEnums),
+    ]),
+};
+
+const defaultProps = {
+    adjacentStickyColumnsTotalWidth: 0,
+    forceTableUpdate: undefined,
+    isResizable: false,
+    resizableColumnWidthPercentage: undefined,
+    stickyColumnCount: undefined,
+    setResizableCellsWordWrapping: undefined,
+    shouldResetWhiteSpaceStyle: true,
+    stickyTableContainerWidth: 0,
+};
+
+const HEADER_CELL_WIDTH_OFFSET = 10;
+const DEBOUNCE_WAIT_TIME = 50;
+export const MAX_WIDTH_PERCENTAGE_DESKTOP = 0.5;
+export const MAX_WIDTH_PERCENTAGE_MOBILE = 0.8;
+
+class TableHeaderCell extends React.PureComponent {
+    constructor() {
+        super();
+        this.state = {
+            minWidth: 0,
+            expandedWidthPercentage: 0,
+            haveExpandedMeasuresBeenSet: false,
+            widthPercentage: 0,
+            shouldDisableResizeHandler: false,
+
+        };
+        this.header = React.createRef();
+        this.debouncedUpdateExpandedWidth = null;
+        this.onClick = this.onClick.bind(this);
+        this.canCellKeepExpanding = this.canCellKeepExpanding.bind(this);
+        this.onDoubleClick = this.onDoubleClick.bind(this);
+        this.onResizeColumn = this.onResizeColumn.bind(this);
+        this.onResizeStop = this.onResizeStop.bind(this);
+        this.getResizingHandler = this.getResizingHandler.bind(this);
+        this.getMaxWidthPercentages = this.getMaxWidthPercentages.bind(this);
+    }
+
+    componentDidMount() {
+        const {
+            isResizable,
+        } = this.props;
+        if (isResizable) {
+            const currentWidth = _.get(this, 'header.current.clientWidth');
+            this.setState({
+                minWidth: currentWidth,
+            });
+        }
+        this.debouncedUpdateExpandedWidth = _.debounce(
+            (initialLoad) => {
+                const {
+                    shouldResetWhiteSpaceStyle,
+                    setResizableCellsWordWrapping,
+                    stickyTableContainerWidth,
+                    resizableColumnWidthPercentage,
+                    adjacentStickyColumnsTotalWidth,
+                    forceTableUpdate,
+                    stickyColumnCount,
+                } = this.props;
+                const {
+                    widthPercentage: currentWidthPercentage,
+                    minWidth,
+                } = this.state;
+                // eslint-disable-next-line max-len
+                const shouldUpdateExpandedWidth = isResizable && minWidth > 0 && shouldResetWhiteSpaceStyle;
+                if (shouldUpdateExpandedWidth) {
+                    // eslint-disable-next-line max-len
+                    const maxWidthPercentages = this.getMaxWidthPercentages(stickyTableContainerWidth);
+                    setResizableCellsWordWrapping(false);
+                    // This ensures the header will re adjust accordint to the content
+                    // in case the column and browser have been already resized
```

Comment:
Nitpicks in the comment text:
* "readjust" can be all one word.
* "according" is misspelled.
* "have been already" should be "have already been"
* I'm not sure what we mean by "the column _and browser_"
* Let's probably end it with a period

Should maybe be something like this...?
```JavaScript
// This ensures that the header will readjust according to
// the content in case the column has already been resized.

```

Thank you ... by the way ... for including many comments around various bits of logic and calculations throughout this body of work in order to explain why things are being done that way.  That is very helpful in my opinion.  👍

Reply:
Updated comment looks good.  Resolving.